code/modules/mining/mine_turfs.dm

```diff
@@ -602,6 +602,17 @@ var/global/list/rockTurfEdgeCache
 	icon_state = "asteroid_dug"
 	return
 
+/turf/simulated/floor/plating/asteroid/singularity_pull(S, current_size)
+	if(current_size == STAGE_THREE)

```

Comment:
This whould get a comment explaining the reason why its overriding the standard floor. Due to the base turf change kor did a while back now.